source/GameInterface/Services/Clans/Patches/ClanChangeInfluencePatch.cs

```diff
@@ -13,37 +13,32 @@ namespace GameInterface.Services.Clans.Patches
     [HarmonyPatch(typeof(ChangeClanInfluenceAction), "ApplyInternal")]
     public class ClanChangeInfluencePatch
     {
-        private static readonly AllowedInstance<Clan> AllowedInstance = new AllowedInstance<Clan>();
-
         static bool Prefix(Clan clan, float amount)
         {
             if (PolicyProvider.AllowOriginalCalls) return true;
 
             if (amount == 0f) return false;
 
-            if (AllowedInstance.IsAllowed(clan)) return true;
+            if (AllowedThread.IsThisThreadAllowed()) return true;
 
             // If not controlled by client skip call
             if (ModInformation.IsClient && clan != Clan.PlayerClan) return false;
 
-            CallStackValidator.Validate(clan, AllowedInstance);
-
             MessageBroker.Instance.Publish(clan, new ClanInfluenceChanged(clan.StringId, amount));
 
             return false;
         }
 
         public static void RunOriginalChangeClanInfluence(Clan clan, float amount)
         {
-            using (AllowedInstance)
-            {
-                AllowedInstance.Instance = clan;
 
-                GameLoopRunner.RunOnMainThread(() =>
+            GameLoopRunner.RunOnMainThread(() =>
+            {
+                using (new AllowedThread())
                 {
                     ChangeClanInfluenceAction.Apply(clan, amount);
-                }, true);
-            }
+                }
+            }, true);
```

Comment:
Can you update this to non-blocking in your branch so we dont forget
```suggestion
            });
```